kratos/python/add_linear_solvers_to_python.cpp

```diff
@@ -237,6 +238,26 @@ void  AddLinearSolversToPython(pybind11::module& m)
     .def("__str__", PrintObject<DeflatedCGSolverType>)
     ;
 
+    using FallbackLinearSolverType = FallbackLinearSolver<SpaceType, LocalSpaceType>;
+    py::class_<FallbackLinearSolverType, FallbackLinearSolverType::Pointer, LinearSolverType>(m, "FallbackLinearSolver")
+    .def(py::init<Parameters>())
+    .def(py::init<LinearSolverType::Pointer, LinearSolverType::Pointer, Parameters>())
+    .def(py::init<const std::vector<LinearSolverType::Pointer>&, Parameters>())
+    .def("AddSolver", [](FallbackLinearSolverType& rSelf, LinearSolverType::Pointer pSolver) {
+        rSelf.AddSolver(pSolver);
+    })
+    .def("AddSolver", [](FallbackLinearSolverType& rSelf, const Parameters ThisParameters) {
+        rSelf.AddSolver(ThisParameters);
+    })
+    .def("GetSolvers", &FallbackLinearSolverType::GetSolvers)
+    .def("SetSolvers", &FallbackLinearSolverType::SetSolvers)
+    .def("GetResetSolverEachTry", &FallbackLinearSolverType::GetResetSolverEachTry)
+    .def("SetResetSolverIndexEachTry", &FallbackLinearSolverType::SetResetSolverIndexEachTry)
+    .def("GetParameters", &FallbackLinearSolverType::GetParameters)
+    .def("GetCurrentSolverIndex", &FallbackLinearSolverType::GetCurrentSolverIndex)
+    .def("ClearCurrentSolverIndex", &FallbackLinearSolverType::ClearCurrentSolverIndex)
```

Comment:
My main usage concern are the setter methods, why do you need those? IMO they are only required if you want to access stuff on a very low level, which none of the usual functionalities like B&S or strategies do.

Same for the other methods, do you add them just because? I would prefer to keep such difficult things easy to use, and extend if there is an actual need

Can you please explain the usecase?

Reply:
For testing and benchmarking mainly

Reply:
I tend to agree @philbucher in here. I'd avoid the `AddSolver` exposition and use them only internally as `private` methods. I cannot figure out a use case in which we may need to add solvers on the fly.

Reply:
This is not solved yet.